### PR TITLE
fix: type regression on fastify

### PR DESCRIPTION
**Pull request by climba03003**
Fixes the type regression on https://github.com/fastify/fastify/pull/3506

### CODE REVIEW

File: pino.d.ts [hunk 1]
```diff
@@ -192,10 +105,98 @@ interface LoggerExtras extends EventEmitter {
 
 declare namespace pino {
     //// Exported types and interfaces
+    
+    interface BaseLogger {
```
Comment (kibertoad) [inline pino.d.ts:109]: why move it from top level into a namespace?

Comment (climba03003) [reply pino.d.ts:109]: Because it never export outside.

Comment (kibertoad) [reply pino.d.ts:109]: can't we export it instead?

Comment (climba03003) [reply pino.d.ts:109]: Then the old way of import is missing this exportation.
e.g.
```
import pino from 'pino'
pino.BaseLogger
```

**Comment by climba03003** [comment]
I have tested even without my changes, it failed the test.

I have no idea why it failed by `@types/jsdom` type check and it is something that we never import.

**Review by mcollina — APPROVED** [review]
lgtm

File: pino.d.ts (hunk 1 above)
Comment (kibertoad) [reply pino.d.ts:109]: We generally want to move away from relying on namespace. See https://github.com/pinojs/pino/issues/1193#issuecomment-966899139
I agree that types should be fixed if we lost the way to do something we could do before, though.

Comment (climba03003) [reply pino.d.ts:109]: I agree that it should move away from namespace, but the current version is too late to do it.
Current version should support the import syntax that can be used in `7.0.0`.

The complete removal should be happen in next major `8.0.0`.

**Comment by mcollina** [comment]
@climba03003 can you rebase? 
@kibertoad fixed this in https://github.com/pinojs/pino/pull/1251.

**Comment by mcollina** [comment]
@kibertoad are you good with this landing?

**Comment by climba03003** [comment]
CI fail is not related to TypeScript changes.

**Comment by mcollina** [comment]
This can land as soon as @kibertoad approves. https://github.com/pinojs/pino/pull/1253 is to improve that test.

File: pino.d.ts (hunk 1 above)
Comment (kibertoad) [reply pino.d.ts:109]: I understand that we can't remove it yet, but can we at least already provide alternative for people who want to do direct named imports? That was the gist of #1193 and it seems like the right direction.
Other than that, PR LGTM.

Comment (climba03003) [reply pino.d.ts:109]: As I mention on line 768, it has the top-level export mapped.

File: pino.d.ts [hunk 2]
```diff
@@ -756,10 +759,13 @@ export type LevelChangeEventListener = pino.LevelChangeEventListener;
 export type LogDescriptor = pino.LogDescriptor;
 export type Logger = pino.Logger;
 export type SerializedError = pino.SerializedError;
+export type SerializerFn = pino.SerializerFn;
 export type SerializedRequest = pino.SerializedRequest;
 export type SerializedResponse = pino.SerializedResponse;
+export type WriteFn = pino.WriteFn;
 
 // Interfaces
+export interface BaseLogger extends pino.BaseLogger {}
```
Comment (climba03003) [inline pino.d.ts:768]: @kibertoad I think here is what you mention?

File: pino.d.ts (hunk 1 above)
Comment (climba03003) [reply pino.d.ts:109]: This PR can import the type I moved inside `namespace` with the below format.

```ts
// named import
import { BaseLogger } from 'pino'
// namespace import / default import
import pino from 'pino'
pino.BaseLogger
// pino named import
import { pino } from 'pino'
pino.BaseLogger
// old P named import
import { P } from 'pino'
P.BaseLogger
```

If anything is missing, please let me know.

**Review by kibertoad — APPROVED** [review]

File: pino.d.ts (hunk 1 above)
Comment (kibertoad) [reply pino.d.ts:109]: Thank you for clarification!

**Pull request merged**